### ISSUE TITLE
Support of other Sphinx extensions

### ISSUE THREAD

**Issue by rainer-steinegger** [trustin/sphinx-gradle-plugin#10]
Hi,

I was wondering what would be the best way to add other Sphinx extensions? 
As far as I understand, the Sphinx runner is pre-built and therefore it is not really possible to add other extensions after it has been released?

Thanks,
Rainer

**Comment by trustin** [comment]
That's correct. It's definitely possible though if we only ship CPython + pip and then build sphinx lazily in a virtual environment.

**Comment by trustin** [comment]
Opening again since I'm interesting making this come true.

**Comment by rainer-steinegger** [comment]
Hi,
Is there any way I could help? I have very little experience with Python and pip and have no idea where I could start :( Therefore pointers would be more than welcome.
Take care,
Raienr

**Comment by trustin** [comment]
Sorry for getting back very late. The idea is:

1. Distribute portable full Python binaries, in a similar way we distribute sphinx-binary.
2. The plugin downloads the Python binary and creates a virtual environment with it.
3. Install various Python modules including sphinx into the virtual environment.

I didn't figure out how to build a portable Python distribution yet, though..

**Comment by trustin** [comment]
One workaround is to put your extensions in `<sphinx_site_dir>/_extensions` and add the following statement to `conf.py`:

```py
sys.path.append(os.path.abspath('_extensions'))
```

**Comment by trustin** [comment]
A working example: https://github.com/line/armeria/tree/armeria-0.68.2/site/src/sphinx

**Comment by trustin** [comment]
@rainer-steinegger I'm curious copying your extensions to your site sources and updating `sys.path` solves your problem.

**Comment by oliva123456** [comment]
adding extensions is possible, but as far as i understand you have to deal with all direct or transitive dependencies manually, which makes including extensions quite erroneous and not so comfortable to do.
e.g. i'm trying to include sphinx-toolbox (https://github.com/sphinx-toolbox/sphinx-toolbox) at the moment and there's the need to add other extensions like apeye, deprecation, deprecation_alias, typing_extensions, ..... 
an alternative is to install the desired extension via python/pip locally and copy the result (from python/lib/site-packages) as extensions.
